src/nanotron/generation/decode.py

```diff
@@ -190,6 +190,16 @@ def decode_text(
 
     p2p = model.p2p
 
+    # replicate input for n_samples times when using TOP_P or TOP_K samplers, in order to get diverse results
+    if generation_config and generation_config.n_samples:
+        if sampler_type != SamplerType.TOP_P and sampler_type != SamplerType.TOP_K:
+            raise ValueError("Only support n_samples for TOP_P and TOP_K sampler")
+        new_input_iter = []
+        for input in input_iter:
+            for _ in range(generation_config.n_samples):
+                new_input_iter.append(GenerationInput(text=input.text))
+        input_iter = new_input_iter
```

Comment:
can you refactor this to be in one line please? (trying to minimize lines of code in src/nanotron 😅)